cloudflare/resource_cloudflare_split_tunnel_exclude.go

```diff
@@ -0,0 +1,127 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareSplitTunnelExclude() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceCloudflareSplitTunnelsExcludeRead,
+		Update: resourceCloudflareSplitTunnelsExcludeUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tunnels": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"address": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Description: "The address for the tunnel.",
+						},
+						"host": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Description: "The domain name for the tunnel.",
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Description: "A description for the tunnel.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCloudflareSplitTunnelsExcludeRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+  splitTunnel, err := client.SplitTunnelExclude(context.Background(), accountID)
+	if err != nil {
+		return fmt.Errorf("Error finding Include Split Tunnels %q", err)
+	}
+
+	tunnelList := make(map[string]interface{}, 0)
+	for _, t := range splitTunnel {
+		tunnelList = append(tunnelList, map[string]interface{}{
+			"address":		  t.Address,
+			"host":			    t.Host,
+			"description":	t.Description,
+		})
+	}
+
+	err = d.Set("tunnels", tunnelList)
+	if err != nil {
+		return fmt.Errorf("error setting Include Split Tunnels: %s", err)
+	}
+	return nil
+}
+
+func resourceCloudflareSplitTunnelsExcludeUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+	// get all of the existing split tunnels
+  existingSplitTunnel, err := client.SplitTunnelExclude(context.Background(), accountID)
+	if err != nil {
+		return fmt.Errorf("Error finding Include Split Tunnels %q", err)
+	}
+
+	tunnelList := make([]interface{}, 0)
+	for _, t := range existingSplitTunnel {
+		tun := cloudflare.SplitTunnel{}
+		if inputAddress, ok := t.GetOk("address"); ok {
+			tun.Address = inputAddress.(string)
+		}
+		if inputHost, ok := t.GetOk("host"); ok {
+			tun.Host = inputHost.(string)
+		}
+		if inputDescription, ok := t.GetOk("description"); ok {
+			tun.Description = inputDescription.(string)
+		}
+
+		tunnelList = append(tunnelList, tun)
+	}
```

Comment:
i see what you're trying to do here but we shouldn't call the API to rebuild payloads for PUT requests. the assumption is that all your data _should_ be in a resource locally (in state) and then you inspect that (`d.Get("tunnels")`) to build the payload. if the local and remote differ, the `Read` method will get them back in sync.